pyani_plus/workflows/__init__.py

```diff
@@ -20,3 +20,173 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 """Module providing snakemake workflows."""
+
+import multiprocessing
+import sys
+from enum import Enum
+from pathlib import Path
+from time import sleep
+
+from rich.progress import Progress
+from sqlalchemy import text
+
+from pyani_plus import db_orm
+from pyani_plus.utils import progress_columns  # rich.progress config
+
+
+class ToolExecutor(str, Enum):
+    """Enum for the executor command line argument passed to snakemake."""
+
+    local = "local"
+    slurm = "slurm"
```

Comment:
In order to define a command line option with predetermined choices in typer, you need to define an enum. I felt the executor enum would be best defined next to all the code for calling snakemake.